src/core/Directus/Database/TableGateway/BaseTableGateway.php

```diff
@@ -798,6 +815,22 @@ protected function executeInsert(Insert $insert)
 
         $resultData = $insertTableGateway->find($generatedValue);
 
+        // try to retrieve the record using another field
+        if (empty($resultData)) {
+            $columnObject = $this->getTableSchema()->getField($this->primaryKeyFieldName);
+            if ($columnObject->hasPrimaryKey()) {
+                $fieldAutoIncrement = null;
+                foreach($this->getTableSchema()->getFields() as $f) {
+                    $fieldAutoIncrement = $f->getName();
+                    $lastInsertValue = $this->getLastInsertValue();
```

Comment:
This block of code doesn't really add anything, because if there is no auto generated id, `$this->getLastInsertValue()` return `0` and the `findOneBy` fails below.